avm/res/event-grid/system-topic/main.bicep

```diff
@@ -53,47 +56,16 @@ var identity = !empty(managedIdentities)
     }
   : null
 
-var builtInRoleNames = {
-  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
-  'EventGrid Contributor': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    '1e241071-0855-49ea-94dc-649edcd759de'
-  )
-  'EventGrid Data Sender': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    'd5a91429-5739-47e2-a06b-3470a27159e7'
-  )
-  'EventGrid EventSubscription Contributor': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    '428e0ff0-5e57-4d9c-a221-2c70d0e0a443'
-  )
-  'EventGrid EventSubscription Reader': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    '2414bbcf-6497-4faf-8c65-045460748405'
-  )
-  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
-  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
-  'Role Based Access Control Administrator': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    'f58310d9-a9f6-439a-9e8d-f62e7b41a168'
-  )
-  'User Access Administrator': subscriptionResourceId(
-    'Microsoft.Authorization/roleDefinitions',
-    '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9'
-  )
-}
-
 var formattedRoleAssignments = [
   for (roleAssignment, index) in (roleAssignments ?? []): union(roleAssignment, {
-    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
```

Comment:
The logic for `builtInRoleNames` still applies. I'd highly recommend using 2 distingt variables. One for the normal role assignments and one for the external ones.